### PR TITLE
feat(metrics): Change rules schema for latest release

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR aims at changing the rules schema for latest release boosting rules.

The new schema uses the `eq` operator and the `null` value instead of `glob` and `not`.

### CODE REVIEW

**Review by andriisoldatenko — APPROVED** [review]

File: src/sentry/dynamic_sampling/rules_generator.py [hunk 1]
```diff
@@ -120,11 +103,18 @@ def generate_boost_release_rules(project_id: int, sample_rate: float) -> List[Re
...
                             "value": [boosted_release.version],
                         },
-                        _generate_environment_condition(boosted_release.environment),
+                        {
+                            "op": "eq",
+                            "name": "trace.environment",
+                            # When environment is None, it will be mapped to equivalent null in json.
+                            # When Relay receives a rule with "value": [null] it will match it against events without
+                            # the environment tag set.
+                            "value": [boosted_release.environment],
```
Comment (jjbayer) [inline src/sentry/dynamic_sampling/rules_generator.py:116]: When the value in the event is `null`, `EqCondition` does not go into the `Array` branch, so I think we need just the value here: 
```suggestion
                            "value": boosted_release.environment,
```

https://github.com/getsentry/relay/blob/a1a12a06f1fdd53566965af4727dc6675268f5b5/relay-sampling/src/lib.rs#L66-L75

Comment (andriisoldatenko) [reply src/sentry/dynamic_sampling/rules_generator.py:116]: good catch!

Comment (iambriccardo) [reply src/sentry/dynamic_sampling/rules_generator.py:116]: This is why I wanted an ingest folk 👀, they know their baby well.

File: src/sentry/dynamic_sampling/rules_generator.py [hunk 2]
```diff
@@ -120,11 +103,20 @@ def generate_boost_release_rules(project_id: int, sample_rate: float) -> List[Re
...
-                        _generate_environment_condition(boosted_release.environment),
+                        {
+                            "op": "eq",
+                            "name": "trace.environment",
+                            # When environment is None, it will be mapped to equivalent null in json.
+                            # When Relay receives a rule with "value": null it will match it against events without
+                            # the environment tag set.
+                            "value": [boosted_release.environment]
+                            if boosted_release.environment
+                            else None,
```
Comment (jjbayer) [inline src/sentry/dynamic_sampling/rules_generator.py:118]:
```suggestion
                            "value": boosted_release.environment
```

`eq` also works with a single value, it then goes into this branch:
https://github.com/getsentry/relay/blob/ae0f47aa5d80bf9086f8bef6575c3713eaf428e2/relay-sampling/src/lib.rs#L68

File: tests/sentry/dynamic_sampling/test_generate_rules.py [hunk 3]
```diff
@@ -379,6 +393,8 @@ def test_generate_rules_with_different_project_platforms(
         },
     ]
     assert generate_rules(default_project) == expected
+    config_str = json.dumps({"rules": generate_rules(default_project)})
+    validate_sampling_configuration(config_str)
```
Comment (jjbayer) [inline tests/sentry/dynamic_sampling/test_generate_rules.py:397]: Thanks for these!

**Review by jjbayer — APPROVED** [review]
I think we can drop the list entirely (see comment). Apart from that, LGTM!

**Pull request merged**